runtime/libia2/include/ia2_internal.h

```diff
@@ -300,9 +300,16 @@ works as a reasonable signpost no-op. */
 #if defined(__aarch64__)
 int ia2_mprotect_with_tag(void *addr, size_t len, int prot, int tag);
 #elif defined(__x86_64__)
+#if LIBIA2_DEBUG_LOG
+static int ia2_mprotect_with_tag(void *addr, size_t len, int prot, int tag) {
+  printf("ia2_mprotect_with_tag(addr=%p, len=%zu, prot=%d, tag=%d)\n", addr, len, prot, tag);
+  return pkey_mprotect(addr, len, prot, tag);
+}
```

Comment:
Maybe we should merge this with the definition for arm in ia2.c?

Reply:
We could, I'm not sure we should have a single function though. I'd rather define separate functions under the conditionals. I'm wondering if it would make sense to make this a static function in the header so we can inline it everywhere?

Reply:
> I'd rather define separate functions under the conditionals.

that part's fine with me

>  I'm wondering if it would make sense to make this a static function in the header so we can inline it everywhere?

Static functions in headers should minimize compartment transitions but they make debugging more complicated since you need to be aware of which copy of the function you're calling and which DSO it's located in. So I'm leaning towards not doing this.